server/routes/product.js

```diff
@@ -0,0 +1,52 @@
+const router = require("express").Router();
+const Product = require("../../server/database/models/product.js");
+
+// route to get all products from db
+router.get("/", (req, res, next) => {
+  Product.findAll()
+    .then(products => res.json(products))
+    .catch(e => {
+      console.error("error in products route", e);
+    });
+});
+
+router.get("/:id", (req, res, next) => {
+  Product.findByPk(req.params.id)
+    .then(product => res.send(product))
+    .catch(e => {
+      console.error("error in single product route", e);
+    });
+});
+
+router.post("/", (req, res, next) => {
+  Product.create(req.body)
+    .then(product => res.json(product))
+    .catch(e => {
+      console.log("error in product.create", e);
+      next(e);
+    });
+});
+
+router.put("/:id", (req, res, next) => {
+  Product.findByPk({
+    where: {
+      id: req.params.id
+    }
+  })
+    .then(product => res.json(product.update(req.body)))
```

Comment:
Is this ever resolving? Need to break this out into another .then()

`.then(product => res.json(product.update(req.body)))`

```javascript
.then(product => product.update(req.body))
.then(updated => res.json(updated)
```